core/src/Kokkos_Core.hpp

```diff
@@ -301,7 +301,7 @@ namespace Experimental {
 //   Customization point for backends
 //   Default behavior is to return the passed in instance
 template <class ExecSpace, class... Args>
-std::vector<ExecSpace> partition_space(ExecSpace space, Args...) {
+std::vector<ExecSpace> partition_space(ExecSpace const& space, Args...) {
```

Comment:
This is fine.  I don't think it matters much and the [documentation](https://github.com/kokkos/kokkos-core-wiki/blob/0de2cca848d11ae62f0d1c8c25bc14b17e866285/API-Reference/Kokkos::partition_space.md#interface) was saying it is passed by const reference.